ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -497,7 +497,7 @@ function admin_ack() {
 
     echo "Require admin ack"
     local wait_time_loop_var=0 ack_data
-    ack_data="$(echo ${out} | awk '{print $2}' | cut -f2 -d\")" && echo "Admin ack patch data is: ${ack_data}"
+    ack_data="$(echo ${out} | awk -F'"' '{print $2}')" && echo "Admin ack patch data is: ${ack_data}"
```

Comment:
I'd like to use `jq` to handle the json output:
```
ack_data="$(echo ${out} | jq -r '. | to_entries[] | .key')"
```

Reply:
Good suggestion. But I'm worried that we'll need to switch to an image without `jq` one day. So I try to keep jsonpath way as far as possible.

Reply:
Firstly, in this script - `cucushift-upgrade-toimage-commands.sh`, there are also other dependences, e.g: ansible, git, I do not think it is easy to be switched to a simple image.
Secondly, if that is really a concern, we can get it without `jq`.
```
oc -n openshift-config-managed get configmap admin-gates -o go-template='{{range $index,$value := .data}}{{printf "%s\n" $index}}{{end}}'
```

Reply:
> Firstly, in this script - cucushift-upgrade-toimage-commands.sh, there are also other dependences, e.g: ansible, git, I do not think it is easy to be switched to a simple image.

ack, then `jq` should be better choice. i think we could get it in line 493 directly then.

```
$ oc -n openshift-config-managed get configmap admin-gates -o json | jq -r ".data|keys[]"
ack-4.12-kube-1.26-api-removals-in-4.13
```

Reply:
I recall that we will get multiple keys in chain upgrade. But we only need to ack one of them in each round of upgrade. cc @JianLi-RH I recall you hit such issues before.

Reply:
> we only need to ack one of them in each round of upgrade

If that, sounds we need to filter out `ack-4.${SOURCE_MINOR_VERSION}` from `oc -n openshift-config-managed get configmap admin-gates -o json | jq -r ".data|keys[]"` output.

Reply:
Ignore my last codes, seem like the `cucushift-chainupgrade-toimage-commands` has the logic about "we only need to ack one of them in each round of upgrade", just need to sync the way of getting the keys there.

Reply:
I meant even the data has been acked in previous upgrade, their keys are still present in the cm. So we need to pick the right one to ack.

Reply:
> even the data has been acked in previous upgrade,

But this case is only existing in `cucushift-chainupgrade-toimage`, right? Not related with this ref step, right?

Reply:
The ref is the last step of chain upgrade. So we will hit the issue if the last upgrade image is stable where adminack is required. Now that seems like all the chain upgrade are on nightly. But I'm thinking to port some of them to stable builds to catch some bugs on upgrade w/o force option.

Reply:
updated~